Ark.Tools.FtpClient.ArxOne/FtpClientPoolArxOneWithSocks.cs

```diff
@@ -54,5 +61,39 @@ private protected override ArxOne.Ftp.FtpClient _getClient()
 
             return client;
         }
+
+        private protected override ArxOne.Ftp.FtpClient _getClientFromUri()
```

Comment:
this is a massive code duplication. 

since ctor either set Uri or Host, a single private and 'if' would suffice.
The ProxyConnect lamda can be made common using a local variable.